agent/pool/pool.go

```diff
@@ -31,7 +32,7 @@ type muxSession interface {
 
 // streamClient is used to wrap a stream with an RPC client
 type StreamClient struct {
-	stream *TimeoutConn
+	stream net.Conn
```

Comment:
This is simply reverting a change from https://github.com/hashicorp/consul/pull/11500